src/components/views/dialogs/InviteDialog.js

```diff
@@ -411,6 +411,58 @@ export default class InviteDialog extends React.PureComponent {
             return scores;
         }, {});
 
+        // Now that we have scores for being in rooms, boost those people who have sent messages
+        // recently, as a way to improve the quality of suggestions. We do this by checking every
+        // room to see who has sent a message in the last few hours, and giving them a score
+        // which correlates to the freshness of their message. In theory, this results in suggestions
+        // which are closer to "continue this conversation" rather than "this person exists".
+        const trueJoinedRooms = client.getRooms().filter(r => r.getMyMembership() === 'join');
+        const now = (new Date()).getTime();
+        const maxAgeConsidered = now - (60 * 60 * 1000); // 1 hour ago
+        const maxMessagesConsidered = 50; // so we don't iterate over a huge amount of traffic
+        const lastSpoke = {}; // userId: timestamp
+        const lastSpokeMembers = {}; // userId: room member
+        for (const room of trueJoinedRooms) {
+            // Skip low priority rooms and DMs
+            const isDm = DMRoomMap.shared().getUserIdForRoomId(room.roomId);
+            if (Object.keys(room.tags).includes("m.lowpriority") || isDm) {
+                continue;
+            }
+
+            const events = room.getLiveTimeline().getEvents(); // timelines are most recent last
+            for (let i = events.length - 1; i >= Math.max(0, events.length - maxMessagesConsidered); i--) {
+                const ev = events[i];
+                if (ev.getSender() === MatrixClientPeg.get().getUserId() || excludedUserIds.includes(ev.getSender())) {
```

Comment:
nit: we're two levels of loop in, we may wish to pull `MatrixClientPeg.get().getUserId()` up to the top level.  That said, I know it's _probably_ just inlined as a lookup anyway, so I don't know if this is worth the effort of changing?

Reply:
it would indeed just be pushing the variable up higher, but I'm not sure if that's much more helpful. We do however already have a `client` variable way up at the top, and use `.getUserId()` a number of times in this function, so will break it up there.

Reply:
and by "a number" I mean "one other time" :|